### PR TITLE
add filter layer side drawer guidance

### PR DESCRIPTION
**Pull request by britt6612**
[Deploy Preview](https://deploy-preview-3637--keen-mayer-a86c8b.netlify.app/components/layer/side-drawer-layer?q=side#filter-layer)

#### What does this PR do?
This PR adds a section about filtering in a side drawer and guidance on how that should be displayed

This also cleans up the code example we had for `Configuration Form` currently it shows [this](https://design-system.hpe.design/components/layer/side-drawer-layer?q=side#configuration-form). This is not only empty but its shows the wrong footer. 
Screen Shot 2024-01-12 at 2 54 55 PM

I changed the code to use the grommet Data too be more aligned. 

#### Where should the reviewer start?
side-drawer-layer.mdx
#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [ ] Cross-browsers (FireFox, Chrome, and Safari)
- [ ] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [ ] Keyboard interactions
- [ ] Screen reader experience
- [ ] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [ ] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### How should this be manually tested?
site
#### Any background context you want to provide?
"Filter layer" should be an exception from other actionable modals.
 update Design System guidance with rationale.
#### What are the relevant issues?
part of issue #3629 
#### Should this PR be mentioned in Design System updates?
yes

### CODE REVIEW

File: aries-site/src/examples/components/menu/MenuBatchActionsExample.js [hunk 1]
```diff
@@ -1,63 +1,74 @@
...
-const filtersConfig = [];
-
 export const MenuBatchActionsExample = ({ containerRef }) => {
   // containerRef is for demonstration purposes on this site. Most
   // implementations should likely remove.
 
   const layerProps = {
     // containerRef is for demonstration purposes on this site. Most
     // implementations should likely remove.
-    target: containerRef && containerRef.current,
```
Comment (britt6612) [inline aries-site/src/examples/components/menu/MenuBatchActionsExample.js:28]: @taysea here is where I have the Layer props lmk your thoughts

Comment (taysea) [reply aries-site/src/examples/components/menu/MenuBatchActionsExample.js:28]: I agree with removing `layerProps` altogether and just doing `<DataFilters layer />` below.

**Review by taysea — COMMENTED** [review]
Looking good, filed a suggestion PR into this branch with some wording thoughts:
https://github.com/grommet/hpe-design-system/pull/3719

File: aries-site/src/examples/components/menu/MenuBatchActionsExample.js (hunk 1 above)
Comment (britt6612) [reply aries-site/src/examples/components/menu/MenuBatchActionsExample.js:28]: same sounds good

**Comment by taysea** [comment]
Screen Shot 2024-02-13 at 11 58 13 AM

Small thing, can we make sure this fills the full width of the example? in mdx maybe we need to add `width="100%"` to Example?

Also, can you add back `DataSummary` below Toolbar?

**Review by taysea — COMMENTED** [review]
The filter layer still has the "Undo changes" button. Can you update the `yarn.lock` to get the latest from Grommet?

**Comment by britt6612** [comment]
@taysea updated now

**Review by taysea — APPROVED** [review]
LGTM, thank you!

**Review by jcfilben — APPROVED** [review]
Looks good!

**Pull request merged**